unittests/AST/ASTImporterTest.cpp

```diff
@@ -1622,13 +1623,49 @@ TEST_F(ImportFunctions,
   }
 
   Decl *ToTU = ToAST->getASTContext().getTranslationUnitDecl();
+  ASSERT_EQ(DeclCounter<FunctionDecl>().match(ToTU, Pattern), 2u);
   FunctionDecl* ProtoD = FirstDeclMatcher<FunctionDecl>().match(ToTU, Pattern);
   EXPECT_TRUE(!ProtoD->doesThisDeclarationHaveABody());
   FunctionDecl* DefinitionD = LastDeclMatcher<FunctionDecl>().match(ToTU, Pattern);
   EXPECT_TRUE(DefinitionD->doesThisDeclarationHaveABody());
   EXPECT_EQ(DefinitionD->getPreviousDecl(), ProtoD);
 }
 
+TEST_F(ImportFunctions,
+       OverriddenMethodsShouldBeImported) {
+  auto Code =
+        R"(
+        struct B { virtual void f(); };
+        void B::f() {}
+        struct D : B { void f(); };
+        )";
+  auto Pattern = cxxMethodDecl(hasName("f"), hasParent(cxxRecordDecl(hasName("D"))));
+  Decl *FromTU = getTuDecl(Code, Lang_CXX);
+  CXXMethodDecl *Proto = FirstDeclMatcher<CXXMethodDecl>().match(FromTU, Pattern);
+
+  ASSERT_EQ(Proto->size_overridden_methods(), 1u);
+  CXXMethodDecl* To = cast<CXXMethodDecl>(Import(Proto, Lang_CXX));
+  EXPECT_EQ(To->size_overridden_methods(), 1u);
+}
+
+TEST_F(ImportFunctions,
```

Comment:
probably add such test not only for the `virtual` flag?

Reply:
We could do that, but I am not sure which flags to check. We had a problem in the previous fix particularly with the virtual flag.

Reply:
Does this test code compile?
Should not `Fixture` be used instead of `ImportFunctions`?

Reply:
`ImportFunctions` came in with commit https://github.com/Ericsson/clang/pull/290/commits/81a877fbc4bd6f0665e17fc313e485967080a0d1